embassy-boot/src/boot_loader.rs

```diff
@@ -49,16 +49,20 @@ pub struct BootLoaderConfig<ACTIVE, DFU, STATE> {
     pub state: STATE,
 }
 
-impl<'a, FLASH: NorFlash>
+impl<'a, ActiveFlash: NorFlash, DFUFlash: NorFlash, StateFlash: NorFlash>
     BootLoaderConfig<
-        BlockingPartition<'a, NoopRawMutex, FLASH>,
-        BlockingPartition<'a, NoopRawMutex, FLASH>,
-        BlockingPartition<'a, NoopRawMutex, FLASH>,
+        BlockingPartition<'a, NoopRawMutex, ActiveFlash>,
+        BlockingPartition<'a, NoopRawMutex, DFUFlash>,
+        BlockingPartition<'a, NoopRawMutex, StateFlash>,
     >
 {
     /// Create a bootloader config from the flash and address symbols defined in the linkerfile
     // #[cfg(target_os = "none")]
-    pub fn from_linkerfile_blocking(flash: &'a Mutex<NoopRawMutex, RefCell<FLASH>>) -> Self {
+    pub fn from_linkerfile_blocking(
```

Comment:
Personally I don't think this API is too bad. Using a builder doesn't provide that much gain, but perhaps a better rustdoc order explaining the parameters would be appropriate

Reply:
Done in 634c409.